app/src/androidTest/java/org/simple/clinic/overdue/AppointmentRepositoryAndroidTest.kt

```diff
@@ -2686,6 +2686,105 @@ class AppointmentRepositoryAndroidTest {
       }
     }
   }
+
+  @Test
+  fun searching_for_overdue_appointments_should_work_correctly() {
+    fun createOverdueAppointment(
+        patientName: String,
+        colonyOrVillageName: String,
+        patientAddressUuid: UUID,
+        patientUuid: UUID,
+        facilityUuid: UUID
+    ) {
+      val patientAddress = testData.patientAddress(
+          uuid = patientAddressUuid,
+          colonyOrVillage = colonyOrVillageName
+      )
+
+      val patientProfile = testData.patientProfile(
+          patientUuid = patientUuid,
+          generatePhoneNumber = true,
+          patientAddressUuid = patientAddressUuid,
+          patientName = patientName
+      )
+
+      patientRepository.save(listOf(patientProfile))
+      database.addressDao().save(listOf(patientAddress))
+
+      val bp = testData.bloodPressureMeasurement(
+          patientUuid = patientUuid,
+          facilityUuid = facilityUuid
+      )
+      bpRepository.save(listOf(bp))
+
+      val appointment = testData.appointment(
+          patientUuid = patientUuid,
+          facilityUuid = facilityUuid,
+          scheduledDate = LocalDate.now().minusDays(10),
+          status = Scheduled,
+          cancelReason = null
+      )
+      appointmentRepository.save(listOf(appointment))
+    }
+
+    // given
+    val searchQuery = "bar"
+    val facilityUuid = UUID.fromString("722111d2-459c-47a6-859c-ca3564fc374d")
+
+    val patient1WithNameMatchingQuery = UUID.fromString("7c7a09ce-d5b7-4c75-9983-01704d8e9cc6")
+    val patient2WithNameMatchingQuery = UUID.fromString("385cf640-ba6d-4fac-8098-612e2620643b")
+    val patientWithVillageNameMatchingQuery = UUID.fromString("0d165e45-d12d-4a4a-bbc4-5f01a7955486")
+    val patientWithNothingMatchingQuery = UUID.fromString("d2046ee2-8474-4202-989d-1f2eac2deb2a")
```

Comment:
Let's remove `Query` suffix from the naming here. This is not a query, so we can just call it `patientWithSomeCondition`

Reply:
That's the condition right? Name matching the search query? which is why I have named it like this

Reply:
So, the condition is just `patientWithMatchingName` or `patient2WithMatchingVillageName`, usually how we name these test patients with different conditions. The query suffix kind of  implies that this is a query rather than an ID.